File: LC1456.cpp
class Solution {
public:
    bool isVowel(char c){
        return c == 'a' || c == 'e' || c == 'i' || c == 'o' || c == 'u';
    }

    int maxVowels(string s, int k) {
        int cnt = 0;

        for(int i = 0; i < k; i++)
            if(isVowel(s[i]))
                cnt++;

        int m = cnt, i = k;

        while(i < s.length()){
            if(isVowel(s[i]))
                cnt++;
            if(isVowel(s[i-k]))
                cnt--;
            m = max(m, cnt);
            i++;
        }

        return m;
    }
};

File: LC1456.py
class Solution:
    def maxVowels(self, s: str, k: int) -> int:
        v = {'a','e','i','o','u'}
        c = 0

        for i in range(k):
            c += int(s[i] in v)
        
        m = c

        for i in range(k, len(s)):
            c += int(s[i] in v) - int(s[i-k] in v)
            m = max(m,c)
        
        return m
        
